docs/framework-overview.md

```diff
@@ -0,0 +1,398 @@
+
+    * handle build options: `Toolchain.prepare_rpath_wrappers`,
+      `Toolchain.prepare_compiler_cache`, `Toolchain.handle_sysroot`
+
+Hence, the base `Toolchain` class provides the skeleton to further develop
+custom toolchain definitions through child classes. EasyBuild already provides
+[many toolchains definitions][toolchains_table] that cover
+[common][common_toolchains] and not-so-common combinations of compilers,
+numeric libraries and MPI implementations. All toolchain definitions are found
+in `easybuild.toolchains`.
+
+Toolchains are modular. Each *toolchain component* (*i.e* compiler, numerical
+library, MPI implementation) is defined by its own child class or by
+multiple ones. These components are combined in sub-toolchains (*e.g.* the
+OpenMPI toolchain) which can then be re-combined with other components in
+bigger toolchains (*e.g.* the FOSS toolchain). Therefore, the resulting options
+of the toolchain, its attributes and methods will be the combined result of all
+inherited class components.
+
+```mermaid
+flowchart
+    toolchain --> compiler & mpi & linalg & fft
+    subgraph "Compiler"
+        compiler --> gcc --> gcccore --> gcctoolchain 
+    end
+    subgraph "MPI Implementation"
+        mpi --> ompi 
+    end
+    gcctoolchain & ompi --> gompi --> foss
+    subgraph "Fast-Fourier Libraries"
+        fft --> fftw 
+    end
+    fftw --> foss
+    subgraph "Linear Algebra Libraries"
+        linalg --> oblas & fblas
+        linalg --> blacs --> scalapack
+    end
+    oblas & fblas & scalapack --> foss
+    toolchain("class Toolchain
+    eb.tools.toolchain.toolchain")
+    compiler("class Compiler
+    eb.tools.toolchain.compiler")
+    gcc("class Gcc
+    eb.toolchains.compiler.gcc")
+    gcccore("class GccCore
+    eb.toolchains.gcccore")
+    gcctoolchain("class GccToolchain
+    eb.toolchains.gcc")
+    mpi("class Mpi
+    eb.tools.toolchain.mpi")
+    ompi("class OpenMPI
+    eb.toolchains.mpi.openmpi")
+    gompi("class Gompi
+    eb.toolchains.gompi")
+    fft("class Fft
+    eb.tools.toolchain.fft")
+    fftw("class Fftw
+    eb.toolchains.fft.fftw")
+    linalg("class LinAlg
+    eb.tools.toolchain.linalg")
+    oblas("class OpenBLAS
+    eb.toolchains.linalg.openblas")
+    fblas("class FlexiBLAS
+    eb.toolchains.linalg.flexiblas")
+    blacs("class Blacs
+    eb.toolchains.linalg.blacs")
+    scalapack("class ScaLAPACK
+    eb.toolchains.linalg.scalapack")
+    foss("class Foss
+    eb.toolchains.foss")
+```
+
+## Features in tools
+
+### EasyBuild Configuration
+
+Configuration options in EasyBuild can come from multiple sources:
+
+* command line arguments to the `eb` command
+* environment variables
+* settings in configuration files
+* settings in easyconfig files
+
+!!! tip
+    Check the [Configuring EasyBuild][configuring_easybuild] for details on how
+    to use the different configuration mechanisms.
+
+All configuration options in EasyBuild are defined in `easybuild.tools.config`.
```

Comment:
It's actually the other way around: all configuration options are defined in `easybuild.tools.options`.

Most of those, but not all, can also be queried via the `build_option`, but only if they're listed in `easybuild.tools.config`